### ISSUE TITLE
Suppress ns.getPlayer warning

### ISSUE THREAD

**Issue by alainbryden** [alainbryden/bitburner-scripts#298]
Because getNsDataThroughFile serializes the entire result of ns.getPlayer - warnings are generated which have already been addressed.

To avoid constant reports of this warning by users, we can inject a little code when this command is detected to omit the deprecated properties from the serializer (https://stackoverflow.com/questions/4910567/hide-certain-values-in-output-from-json-stringify)

**Comment by alainbryden** [comment]
Seems my attempt to suppress the issue didn't work.

**Comment by alainbryden** [comment]
@Arthan-the-one thanks for the suggestion. I got [similar suggestions](https://stackoverflow.com/questions/76373300/serialize-object-without-accessing-certain-keys/76374217#76374217) on stack-overflow and ended up solving it in pretty much that way,

I was hoping to do it all declaratively as a single substituted-statement so I could sneak it in the way I was still doing it, but couldn't quite make it happen, so I settled on using an "immediately invoked function expression" to give me that flexibility without having to mess with the helper function too much.

All good now.
